### ISSUE TITLE
visionOS Support

### ISSUE THREAD

**Issue by PushedCrayon** [apple/swift-nio#2635]
### Expected behavior
Be able to use this package inside a VisionOS application for the Apple Vision Pro

### Actual behavior
Currently appears to fail to compile to the VisionOS simulator

### Steps to reproduce

1. Open the package and try and run to a VisionOS Simulator

### If possible, minimal yet complete reproducer code (or URL to code)

NA

### SwiftNIO version/commit hash

Latest

### System & version information

Running Xcode 15.2 on Sonoma 14.1.2

**Comment by Lukasa** [comment]
Yup, good catch: looks like we need to add some `#if visionOS()`.

**Comment by Lukasa** [comment]
Oh actually before even that swift-system needs to.

**Comment by Lukasa** [comment]
Let's get apple/swift-system#150 fixed first.

**Comment by PushedCrayon** [comment]
I actually forked that one just a bit ago and sprinkled in a lot of `|| os(visionOS)` and it worked just fine. This might be a quick easy win

**Comment by PushedCrayon** [comment]
Looks like https://github.com/apple/swift-system/issues/150 has not been updated for a few years, might have to be something forked and done outside of "official" support

**Comment by Lukasa** [comment]
I mean I only filed that ticket two hours ago.

**Comment by mshur** [comment]
Looks like [apple/swift-system#150](https://github.com/apple/swift-system/issues/150) was marked completed. I think I'm still running into this issue in Nio.  Any update on VisionOS support for Nio?

**Comment by Lukasa** [comment]
Hi @mshur, so far as I know this is working. Can you update your package versions and confirm things are still not compiling? If they aren’t, can you post a build log to show the errors?

**Comment by weissi** [comment]
<img width="881" height="457" alt="Image" src="https://github.com/user-attachments/assets/2126abb9-6a43-4fab-95ea-87f1338aebd5" />

Seems to be okay @mshur , please re-open if that still doesn't work for you
